bodyshopbidsdotcom/git-scripts
# w277901423 Bug: Workers Unable to Pull in us-east-1

Closes https://www.wrike.com/open.htm?id=277901423

## What
This enables us to pass the vpc_id for ECS services to use when updating. This is a short-term fix that unblocks deployments to N.Virginia (us-east-1). Long term fix will require a default VPC to be correctly configured in each region.

This also updates the CPU and MemoryReservation attributes to give containers more resources by default. This is a stop-gap solution until we make this more configurable.

## Why
Services were choosing the VPC we use for Active Directory, due to no default being available. This blocked requests from the ECS Service to other AWS services.

## How
Add a CLI parameter to `git_push_eb`.

## Testing
1. Checkout this git-scripts branch (`w277901423-bug-workers-unable-to-pull-in`)
1. Choose a repository that has workers as ECS Services.
1. Delete the service from ECS (it will be re-created on deploy).
1. Run `git_push_eb --vpc_id vpc-c078d9a9 <ENV NAME>`
1. Navigate to the Service in the ECS cluster for your environment.
1. Look under *Allowed subnets* and confirm that the following values are now under it:
  * subnet-69a21200
  * subnet-9516d3ee
  * subnet-705d443a
1. Delete the service again.
1. Run `git_push_eb <ENV NAME>`
1. Navigate to the ECS Service and confirm that the subnets have been changed to:
  * subnet-ef6c7f97
  * subnet-bcf633d5
  * subnet-343a0a7e
